packages/dds/merge-tree/src/segmentPropertiesManager.ts

```diff
@@ -93,6 +94,11 @@ export class PropertiesManager {
 			// The delta should be null if undefined, as that's how we encode delete
 			deltas[key] = previousValue === undefined ? null : previousValue;
 			const newValue = newProps[key];
+			if (newProps[key] === undefined) {
+				throw new UsageError(
+					"properties cannot contain undefined keys. use null to remove existing keys",
+				);
+			}
```

Comment:
this is what i want to talk about. how to deal with undefined, as today, they do not behave is an eventually consistent manner. the local client will convert to null, but undefined is still serialized, which leads to it not being sent.

I see a few options:

1. Block undefined usages
2. Ensure all undefined get remapped to null in all cases
3. Ignore undefined

This does number 1, but i worry its too strong and could break customers. That said, anyone using this is already broken, but there also may not care, as it may be being done unintentionally, like via object expansion.

Number 2 is really hard due to our layering. And like the above, could cause problems in the unintentional use case.

I didn't like number 3 a first, but it might be the best option, it will break some who is intentionally passing undefined to delete, but this doesn't work anyway. This also seems to better match java script guidance to not differentiate between key existence or not when undefined.